tests/jmap.rs

```diff
@@ -0,0 +1,50 @@
+#![cfg(feature = "invocation")]
+
+extern crate error_chain;
+extern crate jni;
+
+use jni::objects::{
+    JMap,
+    JObject,
+};
+
+mod util;
+use util::{
+    attach_current_thread,
+    unwrap,
+};
+
+#[test]
+pub fn jmap_push_and_iterate() {
+    let env = attach_current_thread();
+    let data = &["hello", "world", "from", "test"];
+
+    let map_object = unwrap(&env, env.new_object("java/util/HashMap", "()V", &[]));
+    let map = unwrap(&env, JMap::from_env(&env, map_object));
+
+    // Push all strings
+    unwrap(
+        &env,
+        data.iter().try_for_each(|s| {
+            env.new_string(s)
+                .map(|s| JObject::from(s))
+                .and_then(|s| map.put(s, s).map(|_| ()))
+        }),
+    );
+
+    let mut collected = Vec::new();
+    unwrap(
+        &env,
+        map.iter().and_then(|mut iter| {
+            iter.try_for_each(|e| {
+                env.get_string(e.0.into())
+                    .map(|s| collected.push(String::from(s)))
```

Comment:
\+ @vitvakatu @DarkEld3r

Is it idiomatic in Rust? In Java streams, for instance, it is encouraged to use collectors instead of manually creating and updating a destination collection: 

```java
List<Result> dst = src.stream()
    .map(Foo::transform)
    .collect(Collectors.toList());
```

Reply:
I did not used `collect` here, because we deal with a lot of `Result`s here. And to not write `unwrap(&env, ...)` every time, I chain all the results.